internal/provider/kubernetes/gateway.go

```diff
@@ -97,28 +103,39 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, request reconcile.Req
 	if acceptedClass == nil {
 		r.log.Info("No accepted gatewayclass found for gateway", "namespace", request.Namespace,
 			"name", request.Name)
-		// TODO: Delete gateway from the IR.
-		// xref: https://github.com/envoyproxy/gateway/issues/38
+		for namespacedName := range r.resourceTable.Gateways.LoadAll() {
```

Comment:
why are we deleting all the gateways here instead of just the one thats missing a valid gateway class ?

Reply:
I think the idea is that if there's no GatewayClass with a matching controller string that has a condition of "Accepted: true", then we should not be programming anything, therefore, we need to ensure all Gateways have been removed from the resource table.